src/Polly.SharedSpecs/Registry/DefaultPolicyRegistrySpecs.cs

```diff
@@ -0,0 +1,238 @@
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(2);
+
+            //Using Indexer
+
+            policy = Policy.NoOp();
+            key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r[key] = policy)
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(3);
+
+            policy = Policy.NoOp();
+            key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r[key] = policy)
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(4);
+        }
+
+        [Fact]
+        public void Should_Allow_Adding_Policy_Using_Indexer()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r[key] = policy)
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(1);
+
+            policy = Policy.NoOp();
+            key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r[key] = policy)
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(2);
+        }
+
+        [Fact]
+        public void Should_Not_Allow_Adding_Policy_With_Duplicate_Key_Using_Add()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldThrow<ArgumentException>();
+
+            _registry.Count.Should().Be(1);
+        }
+
+        [Fact]
+        public void Should_Allow_Retrieving_Stored_Policy_Using_TryGetValue()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            Policy outPolicy = null;
+            bool result = false;
+
+            _registry.Add(key, policy);
+
+            _registry.Invoking(r => result = r.TryGetValue(key, out outPolicy))
+                .ShouldNotThrow<ArgumentException>();
+
+            result.Should().BeTrue();
+
+            outPolicy.ShouldBeEquivalentTo(policy);
+
+            //indexer
+            outPolicy = null;
+            _registry.Invoking(r => outPolicy = r[key])
+                .ShouldNotThrow<KeyNotFoundException>();
+
+            outPolicy.ShouldBeEquivalentTo(policy);
+        }
+
+        [Fact]
+        public void Should_Allow_Retrieving_Stored_Policy_Using_Indexer()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            Policy outPolicy = null;
+
+            _registry.Add(key, policy);
+
+            _registry.Invoking(r => outPolicy = r[key])
+                .ShouldNotThrow<KeyNotFoundException>();
+
+            outPolicy.ShouldBeEquivalentTo(policy);
+        }
+
+        [Fact]
+        public void Should_Not_Throw_While_Retrieving_When_Key_Does_Not_Exist_Using_TryGetValue()
```

Comment:
Added. I also added more test cases to check `ArgumentNullException` is thrown when key is null while adding, removing and retrieving policy.